modules/perception/camera/app/cipv_camera.cc

```diff
@@ -151,42 +151,32 @@ bool Cipv::MakeVirtualLane(const LaneLineSimple &ref_lane_line,
   // TODO(techoe): Use union of lane line and yaw_rate path to define the
   // virtual lane
   virtual_lane_line->line_point.clear();
-  if (!b_image_based_cipv_) {
-    for (uint32_t i = 0; i < ref_lane_line.line_point.size(); ++i) {
-      // Point2Df virtual_line_point(
-      //     ref_lane_line.line_point[i](0),
-      //     ref_lane_line.line_point[i](1) + offset_distance);
-      virtual_lane_line->line_point.emplace_back(
-          ref_lane_line.line_point[i](0),
-          ref_lane_line.line_point[i](1) + offset_distance);
-    }
-  } else {
-    // Image based extension requires to reproject virtual laneline points to
-    // image space.
+  for (uint32_t i = 0; i < ref_lane_line.line_point.size(); ++i) {
+    virtual_lane_line->line_point.emplace_back(
+        ref_lane_line.line_point[i](0),
+        ref_lane_line.line_point[i](1) + offset_distance);
   }
   return true;
 }
 
 float Cipv::VehicleDynamics(const uint32_t tick, const float yaw_rate,
                             const float velocity, const float time_unit,
                             float *x, float *y) {
-  // straight model;
-  *x = time_unit * velocity * static_cast<float>(tick);
-  *y = 0.0f;
-
+  // Option 1. Straight model;
+  // *x = time_unit * velocity * static_cast<float>(tick);
+  // *y = 0.0f;
+
+  // Option 2. Sphere model
+  float adjusted_velocity = std::max(kMinVelocity, velocity);
+  float theta = static_cast<float>(tick) * time_unit * yaw_rate;
+  float displacement =
+      static_cast<float>(tick) * time_unit * adjusted_velocity;
+  *x = displacement * static_cast<float>(cos(theta));
+  *y = displacement * static_cast<float>(sin(theta));
+
+  // Option 3. Bicycle model
```

Comment:
Hi @techoe , Option 3 might be similar to what has been implemented here. https://github.com/ApolloAuto/apollo/blob/master/modules/common/vehicle_model/vehicle_model.cc Not sure if the interface meets your needs through.

Reply:
@Capri2014, Thank you for the information! I did look at the code and that is exactly what I need. I am planning to add it later. For now, since I have to add a routine to read different vehicle parameters and test further, I will make another PR to add the bicycle model. Thank you!